### PR TITLE
feat: remove MongoDB relation

### PR DESCRIPTION
**Pull request by dariofaccin**
# Description

This PR aims to remove the `database` relation from the charm.

# Checklist:

- [x] My code follows the [style guidelines](/CONTRIBUTING.md) of this project
- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that validate the behaviour of the software
- [x] I validated that new and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] I have bumped the version of the library

### CODE REVIEW

**Review by patriciareinoso — CHANGES_REQUESTED** [review]
I thought that we wanted to completely remove the database relation, not to make it optional

File: tests/integration/test_integration.py
Comment (patriciareinoso) [inline tests/integration/test_integration.py:29]: why remove this?

**Review by patriciareinoso — CHANGES_REQUESTED** [review]
looks good. But I don't think we can merge this until the upstream changes are done.
Also, terraform modifications are missing

File: tests/integration/test_integration.py
Comment (dariofaccin) [reply tests/integration/test_integration.py:29]: Markers on fixtures are deprecated since [pytest 7.4](https://docs.pytest.org/en/stable/deprecations.html#applying-a-mark-to-a-fixture-function)

**Review by gatici — COMMENTED** [review]
README.md line 20 `juju integrate sdcore-amf-k8s:database mongodb-k8s`  needs to be removed.  Other than that looks good. We need to wait until upstream changes are merged.

**Review by patriciareinoso — APPROVED** [review]

**Review by gatici — APPROVED** [review]

**Pull request merged**
